custom/logistics/tasks.py

```diff
@@ -101,3 +113,98 @@ def add_products_to_loc(api):
+    # todo: should figure out whether there's a better thing to be doing than faking this global form
+    try:
+        xform = XFormInstance.get(docid='ilsgateway-xform')
+    except ResourceNotFound:
+        xform = XFormInstance(_id='ilsgateway-xform')
+        xform.save()
+    for facility in facilities:
+        sync_stock_transactions_for_facility(domain, endpoint, facility, xform, checkpoint, date, limit, offset)
+        offset = 0  # reset offset for each facility, is only set in the context of a checkpoint resume
+
+
+def sync_stock_transactions_for_facility(domain, endpoint, facility, xform, checkpoint,
+                           date, limit=100, offset=0):
+    """
+    Syncs stock data from StockTransaction objects in ILSGateway to StockTransaction objects in HQ
+    """
+    has_next = True
+    next_url = ""
+    section_id = 'stock'
+    supply_point = facility
+    case = get_supply_point_by_external_id(domain, supply_point)
+    if case:
+        products_saved = set()
+        while has_next:
+            meta, stocktransactions = endpoint.get_stocktransactions(next_url_params=next_url,
+                                                                     limit=limit,
+                                                                     offset=offset,
+                                                                     filters=(dict(supply_point=supply_point,
+                                                                                   date__gte=date,
+                                                                                   order_by='date')))
+
+            # set the checkpoint right before the data we are about to process
+            save_stock_data_checkpoint(checkpoint,
+                                       'stock_transaction',
+                                       meta.get('limit') or limit,
+                                       meta.get('offset') or offset,
+                                       date, facility, True)
+            transactions_to_add = []
+            with transaction.commit_on_success():
+                for stocktransaction in stocktransactions:
+                    params = dict(
+                        form_id=xform._id,
+                        date=force_to_datetime(stocktransaction.date),
+                        type='balance',
+                        domain=domain,
+                    )
+                    try:
+                        report, _ = StockReport.objects.get_or_create(**params)
+                    except StockReport.MultipleObjectsReturned:
+                        # legacy
+                        report = StockReport.objects.filter(**params)[0]
+                    try:
+                        sql_product = SQLProduct.objects.get(code=stocktransaction.product, domain=domain)
+                    except SQLProduct.DoesNotExist:
+                        # todo: kkrampa what's the deal with this logic? this should never be true
```

Comment:
this just moved from somewhere else, but is a good point. @kkrampa why is this check here?

i'd prefer if this didn't block merge but agree it would be good to sort out.